integration/helpers.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+)
+
+func iterOutput[T any](output string, normalize func(*T), compare func(entry *T) error) error {
+	if len(output) == 0 {
+		return fmt.Errorf("output is empty")
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(output))
+	for scanner.Scan() {
+		var entry T
+
+		line := scanner.Text()
+		if err := json.Unmarshal([]byte(line), &entry); err != nil {
+			return fmt.Errorf("unmarshaling line %q: %w", line, err)
+		}
+
+		// To be able to use reflect.DeepEqual and cmp.Diff, we need to
+		// "normalize" the output so that it only includes non-default values
+		// for the fields we are able to verify.
+		if normalize != nil {
+			normalize(&entry)
+		}
+
+		// It returns error when we must stop iterating.
+		if err := compare(&entry); err != nil {
+			return err
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return fmt.Errorf("parsing output: %w", err)
+	}
+
+	return nil
+}
```

Comment:
Something like https://github.com/kinvolk/inspektor-gadget/commit/d9b44b0985a8d76f9db5f31cb33d71881ab88e7b?

Reply:
I didn't used this approach because I wanted to avoid iterating twice over the output. Notice also that both `expectOnly` and `expectAllFrom` will always iterate over all the events. Only `expectAtLeastOne` will return earlier. However, we are not generating to many events, so maybe your suggestion is more suitable in this case. I will change it.